backend/src/main/java/ai/verta/modeldb/utils/UACApisUtil.java

```diff
@@ -232,4 +234,39 @@ public boolean checkAllResourceAllowed(List<Resources> resources) {
             },
             executor);
   }
+
+  public InternalFuture<GetResourcesResponseItem> getEntityResource(
+      String entityId, ModelDBServiceResourceTypes modelDBServiceResourceTypes) {
+    return getResourceItemsForWorkspace(
+            Optional.empty(),
+            Optional.of(Collections.singletonList(entityId)),
+            Optional.empty(),
+            modelDBServiceResourceTypes)
+        .thenApply(
+            responseItems -> {
+              if (responseItems.size() > 1) {
+                var mdbServiceTypeName = modelDBServiceResourceTypes.name();
+                LOGGER.warn(
+                    "Role service returned {}"
+                        + " resource response items fetching {} resource, but only expected 1. ID: {}",
+                    responseItems.size(),
+                    mdbServiceTypeName,
+                    entityId);
+              }
+              Optional<GetResourcesResponseItem> responseItem = responseItems.stream().findFirst();
+              if (responseItem.isPresent()) {
+                return responseItem.get();
+              } else {
+                StringBuilder errorMessage =
+                    new StringBuilder("Failed to locate ")
+                        .append(modelDBServiceResourceTypes.name())
+                        .append(" resources in UAC for ")
+                        .append(modelDBServiceResourceTypes.name())
```

Comment:
These two fields are the same. I believe they probably should be different, right?

Reply:
No, Both are the same one. actual the message would be looks like `Failed to locate PROJECT resources in UAC for PROJECT ID 'ID of resource'` . So there is a used builder pattern to populate that resource name .